app/dashboard/src/providers/ModalProvider.tsx

```diff
@@ -37,23 +39,34 @@ export interface ModalProviderProps extends Readonly<React.PropsWithChildren> {}
 export default function ModalProvider(props: ModalProviderProps) {
   const { children } = props
   const [modal, setModal] = React.useState<Modal | null>(null)
+  // we use key to tell react to invaldidate the modal when we change it.
+  const [key, setKey] = React.useState(0)
```

Comment:
i prefer `useRefresh`, it's true that a numeric `key` does work for any program that runs for less than like a trillion years, but i guess it's just conceptually... less confusing? to not use a number - although it's not like we can't change `useRefresh` to use a number

Reply:
`useRefresh` isn't the same. By setting a key, we tell react that these 2 nodes of `DialogTrigger` are not the same, thus forcing react to remount the component and reinitialize it, and it seems to be correct - since `DialogTrigger` and `Dialog` are 1:1 entities

Reply:
ah right, of course. we have `uniqueString` for that but i guess that's overkill. perhaps we should replace all `uniqueString` to something simpler like this